src/features/calendar.ts

```diff
@@ -0,0 +1,110 @@
+import bot from '../bot'
+import { Telegraf, Context } from 'telegraf'
+import { callbackQuery } from "telegraf/filters"
+import moment from "moment";
+
+bot.command('date', (ctx: Context) => {
+	ctx.reply('Pick a year:', {
+	reply_markup: {
+    	inline_keyboard: generateCalendarKeyboardAnio(),
+    },
+  })
+})
+bot.on(callbackQuery("data"), ctx => {
+    let [actionType, actionValue,days] = ctx.callbackQuery.data.split(':')
+	let date
+    switch (actionType) {
+    	case 'anio':
+        	ctx.answerCbQuery()
+        	ctx.editMessageText('Pick a month:', {
+            	reply_markup: {
+                	inline_keyboard: generateCalendarKeyboardMonth(actionValue),
+                },
+              })
+          break
+        case 'month':
+            ctx.answerCbQuery()
+            ctx.editMessageText('Pick a day:', {
+            	reply_markup: {
+            		inline_keyboard: generateCalendarKeyboardDay(actionValue,days),
+              	},
+            })
+            
+          break
+        case 'day':
+			date = actionValue
+        	ctx.editMessageText(`Date: ${date}`)
+            ctx.answerCbQuery()
+		break
+    }
+
+  })
+function getDate(){
+	bot.on(callbackQuery("data"), ctx => {
+	ctx.reply('Pick a year:', {
+		reply_markup: {
+			inline_keyboard: generateCalendarKeyboardAnio(),
+		},
+	  })
+	})
+}
+function generateCalendarKeyboardAnio() {
+    const keyboard = []
+    const anios = [
```

Comment:
What if we are in 2025?
You don't need to have the same value in different attributes.